mcp-server/src/index.ts

```diff
@@ -0,0 +1,390 @@
+    const uri = request.params.uri;
+    
+    if (uri === 'druid://cluster/status') {
+      const status = await druidClient.getStatus();
+      return {
+        contents: [
+          {
+            uri,
+            mimeType: 'application/json',
+            text: JSON.stringify(status, null, 2),
+          },
+        ],
+      };
+    }
+    
+    if (uri === 'druid://datasources') {
+      const datasources = await druidClient.getDatasources();
+      return {
+        contents: [
+          {
+            uri,
+            mimeType: 'application/json',
+            text: JSON.stringify(datasources, null, 2),
+          },
+        ],
+      };
+    }
+    
+    const datasourceMatch = uri.match(/^druid:\/\/datasource\/(.+)$/);
+    if (datasourceMatch) {
+      const datasource = datasourceMatch[1];
+      const metadata = await druidClient.getDatasourceMetadata(datasource);
+      return {
+        contents: [
+          {
+            uri,
+            mimeType: 'application/json',
+            text: JSON.stringify(metadata, null, 2),
+          },
+        ],
+      };
+    }
+    
+    throw new Error(`Unknown resource URI: ${uri}`);
+  } catch (error) {
+    const errorMessage = error instanceof Error ? error.message : 'Unknown error occurred';
+    throw new Error(`Failed to read resource: ${errorMessage}`);
+  }
+}
+
+// Tool handlers
+async function handleExecuteSqlQuery(request: CallToolRequest): Promise<CallToolResult> {
+  const { query, context } = request.params.arguments as { query: string; context?: any };
+  const result = await druidClient.executeSqlQuery(query, context);
+  
+  return {
+    content: [
+      {
+        type: 'text',
+        text: JSON.stringify(result, null, 2),
+      },
+    ],
+  };
+}
+
+async function handleListDatasources(): Promise<CallToolResult> {
+  const datasources = await druidClient.getDatasources();
+  
+  return {
+    content: [
+      {
+        type: 'text',
+        text: JSON.stringify(datasources, null, 2),
+      },
+    ],
+  };
+}
+
+async function handleGetDatasourceMetadata(request: CallToolRequest): Promise<CallToolResult> {
+  const { datasource } = request.params.arguments as { datasource: string };
+  const metadata = await druidClient.getDatasourceMetadata(datasource);
+  
+  return {
+    content: [
+      {
+        type: 'text',
+        text: JSON.stringify(metadata, null, 2),
+      },
+    ],
+  };
+}
+
+async function handleTestConnection(): Promise<CallToolResult> {
+  const status = await druidClient.getStatus();
+  
+  return {
+    content: [
+      {
+        type: 'text',
+        text: `✅ Connection successful!\n${JSON.stringify(status, null, 2)}`,
+      },
+    ],
+  };
+}
+
+/**
+ * Main server function
+ */
+async function main(): Promise<void> {
+  // Check for help flag
+  if (process.argv.includes('--help') || process.argv.includes('-h')) {
+    showHelp();
+    process.exit(0);
+  }
+
+  console.error('Starting Apache Druid MCP Server...');
```

Comment:
should not use `error`, or the mcp-inspector will recoginize this print as error